src/app/application.cpp

```diff
@@ -907,20 +907,27 @@ int Application::exec()
             QCoreApplication::exit(EXIT_FAILURE);
         connect(m_webui, &WebUI::fatalError, this, []() { QCoreApplication::exit(EXIT_FAILURE); });
 
-        const Preferences *pref = Preferences::instance();
+        printf("%s", qUtf8Printable(u"\n******** %1 ********\n"_s.arg(tr("Information"))));
 
-        const auto scheme = pref->isWebUiHttpsEnabled() ? u"https"_s : u"http"_s;
-        const auto url = u"%1://localhost:%2\n"_s.arg(scheme, QString::number(pref->getWebUiPort()));
-        const QString mesg = u"\n******** %1 ********\n"_s.arg(tr("Information"))
-                + tr("To control qBittorrent, access the WebUI at: %1").arg(url);
-        printf("%s\n", qUtf8Printable(mesg));
+        if (m_webui->isEnabled())
+        {
+            const auto address = m_webui->hostAddress();
+            const auto url = u"%1://%2:%3"_s.arg(m_webui->isHttps() ? u"https"_s : u"http"_s
+                    , (address == QHostAddress::Any) ? u"localhost"_s : address.toString()
```

Comment:
Shouldn't you also take care about `QHostAddress::AnyIPv4` and `QHostAddress::AnyIPv6` in the same way (as I mentioned previously)?

Reply:
Current WebUI code only utilizes `QHostAddress::Any`.

https://github.com/qbittorrent/qBittorrent/blob/24fbaa40b7256f2d252217ae3267e3571ac0b473/src/webui/webui.cpp#L71-L72

So now it basically mimics the original intent and can not end up in the other states.

But if you want to handle all possible cases (e.g. for the future), there is actually quite [more of them](https://doc.qt.io/qt-6/qhostaddress.html#SpecialAddress-enum).
The class also provides a bunch of useful methods we can utilize depending on what do we want from this check.

Reply:
As I can tell `isEqual(QHostAddress::Any, QHostAddress::ConvertUnspecifiedAddress)` covers your suggested case.

> All unspecified addresses will compare equal, namely AnyIPv4, AnyIPv6 and Any.

Reply:
> Current WebUI code only utilizes `QHostAddress::Any`.

You are wrong. WebUI can still listen on `QHostAddress::AnyIPv4` address (if user sets "0.0.0.0" in settings) or on `QHostAddress::AnyIPv6` address (if user sets "::" in settings).

Reply:
> As I can tell `isEqual(QHostAddress::Any, QHostAddress::ConvertUnspecifiedAddress)` covers your suggested case.

Seems Yes. But it needs to be tested and confirmed.

Reply:
> if user sets "0.0.0.0" in settings

I am not sure about `QHostAddress::AnyIPv4 == QHostAddress("0.0.0.0")` resulting to true.
But `isEqual` with the tolerance setting should do the job I guess. I will test it.

Reply:
> I am not sure about `QHostAddress::AnyIPv4 == QHostAddress("0.0.0.0")` resulting to true.

I checked it before writing.

Reply:
Yes, it works. Pushed the changes.

---

Btw, I forgot to mention that getting an actual server port is also necessary because

> If _port_ is 0, a port is chosen automatically.